operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/KubernetesDependentResource.java

```diff
@@ -138,15 +142,23 @@ public R create(R target, P primary, Context<P> context) {
   }
 
   public R update(R actual, R target, P primary, Context<P> context) {
+    if (log.isDebugEnabled()) {
```

Comment:
Why check if debug login is enabled here and not elsewhere?

Reply:
Since creating a new object in that case: `ResourceID.fromResource(actual)`